docs/bytecode.md

```diff
@@ -0,0 +1,208 @@
+# StableHLO Bytecode
+
+## Basic Example
+A minimalist example which only encodes a single attribute in StableHLO can be found [at this commit](https://github.com/openxla/stablehlo/commit/00e3dc98d8d956e5e494be3022df973821e58e91).
+
+There is an image at the bottom of the page which shows difference in the bianry file.
+
+## Currently Encoded Attributes / Types
```

Comment:
Agree. I think more so this section is not needed now that the "### Adding Bytecode for a New Type / Attribute" section exists. This was supposed to be a teaching tool more than anything, but the "how to add" section fills the same role.

Reply:
Will be easier to fiddle with linking after this is merged. I'm thinking to link to a repository search result for "namespace stablehlo_encoding" since that should be reliable, as the enum declarations for attrs/types is the only thing in that namespace.

Reply:
"Will be easier to fiddle with linking after this is merged". Sounds good.